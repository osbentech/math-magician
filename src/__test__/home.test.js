const { render, screen } = require('@testing-library/react');
require('@testing-library/jest-dom');
const Home = require('../home').default;

describe('Tests for Home Component', () => {
  it('Renders Home page', () => {
    const home = render(<Home />);
    expect(home).toMatchSnapshot();
  });

  it('Test if the welcome header exists', () => {
    render(<Home />);
    expect(
      screen.getByText(
        /Welcome to my homepage!/i,
      ),
    ).toBeInTheDocument();
  });
});
