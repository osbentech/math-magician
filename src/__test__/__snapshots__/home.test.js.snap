// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Home Component Renders Home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header>
          <h1>
            A Benjamin think...
          </h1>
          <ul>
            <a
              href="/"
            >
              Home
            </a>
            <span>
               | 
            </span>
            <a
              href="/Calculator"
            >
              Calculator
            </a>
            <span>
               | 
            </span>
            <a
              href="/Quote"
            >
              Quotes
            </a>
          </ul>
        </header>
        <main>
          <h2>
            Welcome to my homepage!
          </h2>
          <p>
            est vitae ex. Duis nisi lacus, facilisis in varius ac, pulvinar nec libero. Pellentesque feugiat elementum bibendum. Ut convallis libero sit amet tortor egestas, nec laoreet eros ullamcorper. Nullam laoreet semper rutrum. Mauris leo enim, vestibulum ut lacus sed, blandit imperdiet nibh. Donec eget velit nec ipsum auctor egestas. Nullam eget lacinia sapien. Vivamus faucibus ut nibh vel tristique. Aliquam rutrum eros a augue malesuada, a egestas tortor rhoncus Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla dignissim, dolor id sagittis tincidunt, mauris mauris consequat leo, et commodo purus est vitae ex. Duis nisi lacus, facilisis in varius ac, pulvinar nec libero. Pellentesque feugiat elementum bibendum. Ut convallis libero sit amet tortor egestas, nec laoreet eros ullamcorper. Nullam laoreet semper rutrum. Mauris leo enim, vestibulum ut lacus sed, blandit imperdiet nibh. Donec eget velit nec ipsum auctor egestas. Nullam eget lacinia sapien. Vivamus fauc
          </p>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header>
        <h1>
          A Benjamin think...
        </h1>
        <ul>
          <a
            href="/"
          >
            Home
          </a>
          <span>
             | 
          </span>
          <a
            href="/Calculator"
          >
            Calculator
          </a>
          <span>
             | 
          </span>
          <a
            href="/Quote"
          >
            Quotes
          </a>
        </ul>
      </header>
      <main>
        <h2>
          Welcome to my homepage!
        </h2>
        <p>
          est vitae ex. Duis nisi lacus, facilisis in varius ac, pulvinar nec libero. Pellentesque feugiat elementum bibendum. Ut convallis libero sit amet tortor egestas, nec laoreet eros ullamcorper. Nullam laoreet semper rutrum. Mauris leo enim, vestibulum ut lacus sed, blandit imperdiet nibh. Donec eget velit nec ipsum auctor egestas. Nullam eget lacinia sapien. Vivamus faucibus ut nibh vel tristique. Aliquam rutrum eros a augue malesuada, a egestas tortor rhoncus Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla dignissim, dolor id sagittis tincidunt, mauris mauris consequat leo, et commodo purus est vitae ex. Duis nisi lacus, facilisis in varius ac, pulvinar nec libero. Pellentesque feugiat elementum bibendum. Ut convallis libero sit amet tortor egestas, nec laoreet eros ullamcorper. Nullam laoreet semper rutrum. Mauris leo enim, vestibulum ut lacus sed, blandit imperdiet nibh. Donec eget velit nec ipsum auctor egestas. Nullam eget lacinia sapien. Vivamus fauc
        </p>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
